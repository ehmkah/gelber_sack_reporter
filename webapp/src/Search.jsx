import React, { Component } from 'react';
import preload from './data.json';

import AbholungDetail from './AbholungDetail';
import ZipCodeLoader from './ZipCodeLoad';

class Search extends Component {
  state = {
    streetName: 'laue',
      zipCode: ''

  };

  handleStreetnameChange = event => {
    this.setState({ streetName: event.target.value });
  };

  handleZipCodeChange = event => {
      this.setState({zipCode: event.target.value});
  }

  render() {
    return (
      <div>
          <input
              onChange={this.handleZipCodeChange}
              value={this.state.zipCode}
              type="text"
          />
        <input
          onChange={this.handleStreetnameChange}
          value={this.state.streetName}
          type="text"
        />
        <div>
           < ZipCodeLoader zipCode={this.state.zipCode}/>



          {preload.d.results
            .filter(
              abholung =>
                abholung.Street.toUpperCase().indexOf(
                  this.state.streetName.toUpperCase()
                ) >= 0 && abholung.Houseno === ''
            )
            .map(abholung => <AbholungDetail street={abholung.Street} adrKey={abholung.Adrkey} />)}
        </div>
      </div>
    );
  }
}

export default Search;
