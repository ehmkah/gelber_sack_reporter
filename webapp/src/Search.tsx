import React, { Component } from "react";

interface State {
  zipCode: string;
}

class Search extends Component<object, State> {
  public state = {
    zipCode: "1234"
  };

  public foo() {
    const newVal: State = { zipCode: "4561" };
    this.setState(newVal);
  }

  public render() {
    return <div>hallo</div>;
  }
}

export default Search;
export { State };
