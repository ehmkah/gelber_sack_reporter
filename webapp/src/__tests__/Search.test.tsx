import React from "react";
import { create } from "react-test-renderer";
import { shallow } from "enzyme";
import Search, { State } from "../Search";

test("snapshot", () => {
  const c = create(<Search />);
  expect(c.toJSON()).toMatchSnapshot();
});

test("example unit test", () => {
  const wrapper = shallow(<Search />);
  const state = wrapper.state() as State;
  expect(state.zipCode).toBe("1234");
  const seacrh = wrapper.instance() as Search;
  seacrh.foo();
  const state2 = wrapper.state() as State;
  expect(state2.zipCode).toBe("4561");
});
