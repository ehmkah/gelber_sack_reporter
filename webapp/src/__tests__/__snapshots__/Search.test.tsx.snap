// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Postleitzahl"
    type="text"
    value=""
  />
  <div />
  <div>
    noch nicht geladen
  </div>
</div>
`;
